### DISCUSSION TITLE
Battery level not dropping

### DISCUSSION THREAD

**Discussion by Corythosaurs** [rlogiacco/BatterySense#46] [Q&A]
hey, I'm very new to all this so this mistake might be very obvious. I have a 12v NiMH battery hooked up to a 10k resistor and 20k resistor. But according to the code the battery level is not dropping and its still reading 100% even when i unplug the battery is there a reason why?

code:
```
#include "Battery.h"

Battery battery(10700, 12000, A0);
/**
 * 1 cell li-ion/li-poly battery wired to A0 before voltage booster, on demand sensing on pin 3, linear mapping function
 * https://github.com/rlogiacco/BatterySense#lesser-than-5v-with-voltage-booster
 **/
void setup() {
	Serial.begin(9600);
  float ratio = (22+10) / 10;
	while (!Serial);
	battery.begin(5000, 3.);
}

void loop() {
	digitalWrite(13, HIGH);
	delay(500);
	digitalWrite(13, LOW);
	delay(500);

	Serial.print("Battery voltage is ");
	Serial.print(battery.voltage());
	Serial.print(" (");
	Serial.print(battery.level());
	Serial.println("%)");
}
```
![IMG_7604](https://github.com/rlogiacco/BatterySense/assets/139459753/28b1d3ec-7dd8-4a4d-a342-27890f4f8cbf)

**Comment by NuclearPhoenixx** [comment]
Well, if you unplug your battery, but still are using the microcontroller, I'm guessing you've connected it externally, right? Then your battery charger is probably outputting a charging voltage which your ADC is picking up despite there not being a battery.

**Reply by Corythosaurs** [reply]
yea but, i dont think the sensor port is detecting voltage from the type b port.

**Reply by NuclearPhoenixx** [reply]
What are you talking about?

**Reply by Corythosaurs** [reply]
the data port, anyways the issue seemed to have solved itself, but the fluctuation is crazy

**Reply by rlogiacco** [reply]
What do you mean by "the fluctuation is crazy"? Is it the voltage or the percentage and what type of swing are you experiencing? Remember, an analog pin acts like a tiny internal capacitor: the capacitor is connected to the input, charged, then disconnected and then the voltage of the capacitor is measured. 
The wiring you are showing in the picture with the attached code should not swing: you are using one single analog pin not absorbing a lot of current prior to the measurement...

**Reply by Corythosaurs** [reply]
The voltage and percentage are fluctuating, as in the voltage could drop from 90% to 70% if I so much as touch the pins, I'm not sure if this is a problem with the breadboard or bad soldering on my part or something else.

**Reply by rlogiacco** [reply]
None of the above: your body is a capacitor, if you touch the pins you charge up, briefly lowering the voltage. It's this the principle used by your smartphone touchscreen...

**Reply by Corythosaurs** [reply]
No, it will stay at 70% after I touched it

**Reply by rlogiacco** [reply]
That is weird... do you have a multimeter to check the voltage values before and after? A 20% drop is quite significant and should only be transient. I would also compare the voltage at the junction before and after, trying to identify the source: breadboards, especially the cheap ones, can be the source of many troubles, but this is quite exaggerated even for a cheap Chinese breadboard...

**Comment by rlogiacco** [comment]
> hey, I'm very new to all this so this mistake might be very obvious. I have a 12v NiMH battery hooked up to a 10k resistor and 20k resistor. But according to the code the battery level is not dropping and its still reading 100% even when i unplug the battery is there a reason why?
> 
> code:
> 
> ```
> #include "Battery.h"
> 
> Battery battery(10700, 12000, A0);
> /**
>  * 1 cell li-ion/li-poly battery wired to A0 before voltage booster, on demand sensing on pin 3, linear mapping function
>  * https://github.com/rlogiacco/BatterySense#lesser-than-5v-with-voltage-booster
>  **/
> void setup() {
> 	Serial.begin(9600);
>   float ratio = (22+10) / 10;
> 	while (!Serial);
> 	battery.begin(5000, 3.);
> }
> 
> void loop() {
> 	digitalWrite(13, HIGH);
> 	delay(500);
> 	digitalWrite(13, LOW);
> 	delay(500);
> 
> 	Serial.print("Battery voltage is ");
> 	Serial.print(battery.voltage());
> 	Serial.print(" (");
> 	Serial.print(battery.level());
> 	Serial.println("%)");
> }
> ```

First of all, is the 10k resistor the upper or lower of the two? It should be the lower one.
The ratio you specified is slightly off, should be `3.3` rather than `3.`.

If you unplug the battery and use a multimeter you will see the Vin pin is not at 0, but at `5V`, the voltage it gets from the USB port: this means the voltage you see at the resistor junction is always going to be 3.3 times the Vin.

How to fix: use some electrical tape to isolate the power pin of the USB connector, so that the USB is not providing power to the Uno, I would also add a diode between the battery and Vin (this should also help with inverted polarity).

In any case,  you should use a multimeter to see the voltage at the resistors' junction: that's what the library is using to determine the voltage and, as a consequence, the battery charge level.

**Reply by rlogiacco** [reply]
> always going to be 3.3 times the Vin

sorry, I meant 1/3.3 times the Vin, approx one-third, which should not give a 100% percentage anyway.

**Reply by Corythosaurs** [reply]
>  Vin pin is not at 0, but at `5V`

I do, this might solve it thanks.
